### DISCUSSION TITLE
Agents need to be defined in both task and crew. Are both required? What is the role of an agent defined in crew

### DISCUSSION THREAD

**Discussion by Frog863** [crewAIInc/crewAI#1179] [Q&A]
######Example:
task2 = Task(
  description="""Using the insights provided, develop an engaging blog
  post that highlights the most significant AI advancements.
  Your post should be informative yet accessible, catering to a tech-savvy audience.
  Make it sound cool, avoid complex words so it doesn't sound like AI.""",
  expected_output="Full blog post of at least 4 paragraphs",
  agent=writer
)
crew = Crew(
  agents=[researcher, writer],
  tasks=[task1, task2],
  verbose=2, # You can set it to 1 or 2 to different logging levels
)
We can see agents both apear in Crew and Task

**Comment by theCyberTech** [comment]
the agents and takes are added as parameters to the crew instance, this is how it should work

In a task the agent is an expected attribute which links the task to the agent

**Comment by cbruyndoncx** [comment]
The agent attribute at the task is optional, not mandatory
